apps/examples/src/examples/export-canvas-as-image/ExportCanvasImageExample.tsx

```diff
@@ -9,17 +9,13 @@ function ExportCanvasButton() {
 			onClick={async () => {
 				const shapeIds = editor.getCurrentPageShapeIds()
 				if (shapeIds.size === 0) return alert('No shapes on the canvas')
-				const blob = await exportToBlob({
-					editor,
-					ids: [...shapeIds],
-					format: 'png',
-					opts: { background: false },
-				})
+				const { blob } = await editor.toImage([...shapeIds], { format: 'png', background: false })
 
 				const link = document.createElement('a')
-				link.href = window.URL.createObjectURL(blob)
+				link.href = URL.createObjectURL(blob)
 				link.download = 'every-shape-on-the-canvas.jpg'
 				link.click()
+				URL.revokeObjectURL(link.href)
```

Comment:
for some reason this makes me worried: is this consistent across browsers in working? i.e. will the link work before this is revoked? i'm worried there's some timing issue potential here.
maybe try in a couple browsers with a large 10+MB image to be sure.

Reply:
under the hood, object URLs like this work sort of like refcounted references to the underlying blob. ie as long as a URL is created but not revoked, the blob can't be garbage collected. Once the download is started, the download stream from the blob is the new reference stopping it from getting garbage collected. Revoking the object URL won't abort the stream (there are a few examples from around the web of people testing this out)